beamer/state_machine.py

```diff
@@ -61,6 +61,16 @@ class Context:
     l1_invalidations: dict[ClaimId, Future]
     finality_periods: dict[ChainId, int] = field(default_factory=dict)
 
+    @property
+    def source_rpc_url(self) -> URL:
+        provider = cast(HTTPProvider, self.request_manager.web3.provider)
+        return URL(str(provider.endpoint_uri))
```

Comment:
Why do we need `str` here? Wouldn't just `URL(provider.endpoint_uri)` be enough?

Reply:
So why you don't need a call to `str` in `conftest.py`?

Reply:
imo `brownie.web3.provider.endpoint_uri` is also wrong as provider is BaseProvider but we dont have ignore there or we dont cast it. as endpoint_uri in conftest is Any i guess it is not raising anything but here we cast it and endpoint_uri is URI | None and it is raising.

Reply:
and without casting it is raising here for source_rpc_url

Reply:
only allowed to use `URL(provider.endpoint_uri or "")` if it is ok or need to use str directly.